external-builds/rockbuilder.py

```diff
@@ -0,0 +1,230 @@
+        print("Alternatively activate the virtual python environment")
+        sys.exit(1)
+
+print("ROCM_HOME: " + os.environ["ROCM_HOME"])
+print("ROCK_PYTHON_HOME: " + os.environ["ROCK_PYTHON_HOME"])
+print("ROCK_BUILDER_HOME_DIR: " + os.environ["ROCK_BUILDER_HOME_DIR"])
+print("ROCK_BUILDER_SRC_DIR: " + os.environ["ROCK_BUILDER_SRC_DIR"])
+print("ROCK_BUILDER_BUILD_DIR: " + os.environ["ROCK_BUILDER_BUILD_DIR"])
+print("PATH: " + os.environ["PATH"])
+print(ENV_VARIABLE_NAME__LIB + ": " + os.environ[ENV_VARIABLE_NAME__LIB])
+time.sleep(1)
+
+# Create an ArgumentParser object
+parser = argparse.ArgumentParser(description='ROCK Project Builders')
+
+# Add arguments
+parser.add_argument('--project', type=str, help='select target for the action. Can be either one project or all projects in core_apps.cfg.', default='all')
+parser.add_argument('--checkout',  action='store_true', help='checkout source code for the project', default=False)
+parser.add_argument('--clean',  action='store_true', help='clean build files', default=False)
+parser.add_argument('--configure',  action='store_true', help='configure project for build', default=False)
+parser.add_argument('--build',  action='store_true', help='build project', default=False)
+parser.add_argument('--install',  action='store_true', help='install build project', default=False)
+
+# Parse the arguments
+args = parser.parse_args()
+
+if ("--checkout" in sys.argv) or ("--clean" in sys.argv) or ("--configure" in sys.argv) or\
+   ("--build" in sys.argv) or ("--install" in sys.argv):
+    print("checkout/clean/configure/build or install argument specified")
+else:
+    print("no checkout/clean/configure/build or install argument specified")
+    print("assuming all of them are enabled")
+    args.checkout=True
+    args.configure=True
+    args.build=True
+    args.install=True
+
+# Access the arguments
+print('project:', args.project)
+print('checkout:', args.checkout)
+print('clean:', args.clean)
+print('configure:', args.configure)
+print('build:', args.build)
+print('install:', args.install)
+
+project_manager = project_builder.RockExternalProjectListManager()
+# allow_no_value param says that no value keys are ok
+sections = project_manager.sections()
+print(sections)
+project_list = project_manager.get_external_project_list()
+print(project_list)
+
+# checkout all projects
+if args.checkout:
+    if (args.project == "all"):
+        for ii, prj_item in enumerate(project_list):
+            print(f"checkout index: {ii}, project: {prj_item}")
+            prj_builder = project_manager.get_rock_project_builder(project_list[ii])
+            if (prj_builder is None):
+                sys.exit(1)
+            else:
+                prj_builder.printout()
+                prj_builder.checkout()
+    else:
+        prj_builder = project_manager.get_rock_project_builder(args.project)
+        if (prj_builder is None):
+            sys.exit(1)
+        else:
+            prj_builder.printout()
```

Comment:
For the code review, can you share some logs (e.g. in a https://gist.github.com/) of this script running? I'd like to see how the steps all fit together. Maybe a trimmed snippet, if the build logs from e.g. PyTorch are too long.

Reply:
Nice! Looking good!